dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -322,14 +349,93 @@ object Engine {
       }
       Some(sc)
     } else {
-      physicalCoreNumber = cores
+      _onSpark = false
```

Comment:
The default value is half of the virtual core in the OS. User can change it through DL_CORE_NUMBER env variable.